src/client/components/network-editor/right-drawer.js

```diff
@@ -0,0 +1,184 @@
+
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    width: RIGHT_DRAWER_WIDTH,
+    background: chroma(theme.palette.background.default).alpha(0.66).hex(),
+    backdropFilter: 'blur(8px)',
+    borderLeft: `1px solid ${theme.palette.divider}`,
+  },
+  header: {
+    flex: '0 1 auto',
+  },
+  toolbar: {
+    paddingLeft: theme.spacing(1),
+    paddingRight: theme.spacing(0.5),
+    minHeight: HEADER_HEIGHT,
+  },
+  content: {
+    flex: '1 1 auto',
+    overflowY: 'auto',
+    borderColor: theme.palette.divider,
+    borderWidth: '1px',
+    borderStyle: 'solid solid hidden hidden',
+    borderRadius: 4,
+  },
+  title: {
+    paddingLeft: theme.spacing(0.5),
+  },
+  grow: {
+    flexGrow: 1,
+  },
+  closeButton: {
+    width: 41,
+    height: 41,
+  },
+}));
+
+
+const RightDrawer = ({ menu, open, onClose }) => {
+  const [ menuStack, setMenuStack ] = useState([ menu ]);
+  const [ forward, setForward ] = useState(true);
+  const [ exit, setExit ] = useState(false);
+
+  const exitedRef = useRef(true); // Will indicate whether or not the slide transition has exited
+  const forwardSubMenu = useRef(null); // Save the next sub-menu to be rendered after exiting the transition
+  const goForwardRef = useRef(null);
+  const goBackwardRef = useRef(null);
+  
+  const classes = useStyles();
+
+  useEffect(() => {
+    if (!open) { // Reset to original state when the menu is closed...
+      setMenuStack([ menu ]);
+    }
+  }, [open]);
+
+  const goBackward = () => {
+    if (exitedRef.current) {
+      // The transition has already exited, so it can enter again...
+      if (menuStack.length > 1) {
+        setExit(false); // Reset this flag
+        setForward(false);
+        menuStack.pop();
+        setMenuStack([ ...menuStack ]);
+      }
+    } else {
+      // The transition has not exited yet, so just force it to exit first...
+      goBackwardRef.current = goBackward; // ...but save the next function to be called on the exited event
+      goForwardRef.current = null;
+      forwardSubMenu.current = null;
+      setExit(true);
+    }
+  };
+  const goForward = (subMenu) => {
+    if (exitedRef.current) {
+      // The transition has already exited, so it can enter again...
+      if (subMenu) {
+        setExit(false); // Reset this flag
+        setForward(true);
+        menuStack.push(subMenu);
+        setMenuStack([ ...menuStack ]);
+      }
+    } else {
+      // The transition has not exited yet, so just force it to exit first...
+      goBackwardRef.current = null;
+      goForwardRef.current = goForward; // ...but save the next function to be called on the exited event
+      forwardSubMenu.current = subMenu; // ...and the next sub-menu
+      setExit(true);
+    }
+  };
+  const handleClick = (fn, subMenu) => {
+    if (subMenu) {
+      goForward(subMenu);
+    } else if (fn) {
+      onClose();
+      fn();
+    }
+  };
+
+  // The transition must always exit before it enters again,
```

Comment:
I managed to make the `Slide` transition work (when animating from a menu to a submenu and vice-versa) by waiting for an 'exit' event before causing the `Slide` to 'enter' again. So, on each re-render, if not exited yet, we save the next function to be called (`goForward` or `goBackward`) and then tell the `Slide` to exit (see `setExit(true)` and `in={!exit}`). Finally, if exited, it executes the saved function, which also causes the Slide to 'enter' (i.e. run the animation--see `setExit(false)`).
I found this solution quite complicated, so feel free to play with this and maybe suggest a better alternative, if you want to.